feat: Implement YAML config parser, CLI tool, and GitHub Action

salihdev0 opened the pull request:
Closes #2

This commit adds the following features:

1. YAML Configuration Parser:
   - Create a new package for configuration (pkg/config)
   - Support for all rule types (layer dependencies, interface implementations, parameter types)
   - Validation for configuration files

2. CLI Tool Implementation:
   - Create a cmd directory with main.go
   - Implement CLI functionality to read and parse YAML config
   - Add verbose output for all rule types

3. GitHub Action Implementation:
   - Create action.yml file
   - Implement Dockerfile for the action
   - Add example workflow file

4. Example Configurations:
   - Basic configuration with layer dependencies
   - Advanced configuration with interface and parameter rules
   - Layer-specific configuration
   - Hexagonal architecture configuration
   - Clean architecture configuration

All features have been tested and are working as expected.

salihdev0 committed:
feat: Implement YAML config parser, CLI tool, and GitHub Action

This commit adds the following features:

1. YAML Configuration Parser:
   - Create a new package for configuration (pkg/config)
   - Support for all rule types (layer dependencies, interface implementations, parameter types)
   - Validation for configuration files

2. CLI Tool Implementation:
   - Create a cmd directory with main.go
   - Implement CLI functionality to read and parse YAML config
   - Add verbose output for all rule types

3. GitHub Action Implementation:
   - Create action.yml file
   - Implement Dockerfile for the action
   - Add example workflow file

4. Example Configurations:
   - Basic configuration with layer dependencies
   - Advanced configuration with interface and parameter rules
   - Layer-specific configuration
   - Hexagonal architecture configuration
   - Clean architecture configuration

All features have been tested and are working as expected.

mstrYoda commented on pkg/config/config_test.go:
```diff
@@ -0,0 +1,184 @@
+
+rules:
+  - from: Application
+    to: Domain
```
May be we can use the same naming convention for here too.

Instead of "to" what about using "DependsOn" ?

salihdev0 committed:
refactor: improve naming conventions in YAML configuration

This commit improves the naming conventions in the YAML configuration structure:

1. Changed field names to be more consistent with the API:
   - `from` -> `layer`
   - `to` -> `dependsOn`
   - `sourceLayer` -> `layer`
   - `targetLayer` -> `dependsOn`
   - `allowed` -> `isAllowed`
   - `shouldUseInterface` -> `useInterface`
   - `ruleType` -> `type`

2. Updated all example configurations to use the new field names

3. Updated CLI tool to display the new field names in verbose output

4. Updated tests to use the new field names

These changes make the configuration more intuitive and consistent with the
method names in the API (e.g., DependsOnLayer).
